src/debt-total.js

```diff
@@ -0,0 +1,87 @@
+'use strict';
+
+var calcDebt = require( './calc-debt.js' );
+
+ /**
+  * calculate total debt
+  * @param { object } data - the data object
+  * @returns { object } the data object with total debt
+  */
+function debtTotal( data ) {
+
+  // Borrowing over cost of attendance
+  data.overborrowing = 0;
+  if ( data.yearOneCosts < data.outofpockettotal +
+    data.borrowingtotal ) {
+    data.overborrowing = data.borrowingtotal +
+                         data.outofpockettotal -
+                         data.yearOneCosts;
+  }
+
+  // Estimated Debt Calculation
+  // Perkins debt at graduation
+  data.perkinsTotal = data.perkins * data.programLength;
+
+  // Direct Subsidized Loan with 1% Origination Fee
+  data.staffSubsidizedWithFee = data.staffSubsidized *
+                                data.DLOriginationFee;
+
+  // Subsidized debt at graduation
+  data.staffSubsidizedTotal = data.staffSubsidizedWithFee *
+                              data.programLength;
+
+  // Direct Unsubsidized Loan with 1% Origination Fee
+  data.staffUnsubsidizedWithFee = data.staffUnsubsidized *
+                                  data.DLOriginationFee;
+
+  // Unsubsidized debt at graduation
+  data.staffUnsubsidizedTotal = calcDebt(
+    data.staffUnsubsidizedWithFee,
+    data.unsubsidizedRate,
+    data.programLength,
+    data.deferPeriod
+  );
+
+  // Grad Plus with origination
+  data.gradplusWithFee = data.gradplus * data.plusOriginationFee;
+
+  // Grad Plus debt at graduation
+  data.gradplusTotal = calcDebt(
+    data.gradplusWithFee,
+    data.gradplusRate,
+    data.programLength,
+    data.deferPeriod
+  );
+
+  // Parent Plus Loans with origination fees
+  data.parentplusWithFee = data.parentplus * data.plusOriginationFee;
```

Comment:
Great question and that certainly makes sense to me. I haven't touched any of the calculations, just moved things around and cleaned them up. I've been collecting a list of similar questions for Patrick and I'll add this to the list.